samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/bean/delivery/DeliveryBean.java

```diff
@@ -1965,9 +1965,9 @@ public boolean getShowFeedbackLink() {
       return ("reviewAssessment".equals(getActionString())
               || "takeAssessment".equals(getActionString())
               || "takeAssessmentViaUrl".equals(getActionString())
-              || "previewAssessment".equals(getActionString())
-             ) && !"1".equals(getNavigation())
-               && getPageContents().getIsNoParts();
+              || "previewAssessment".equals(getActionString()))
+             && ! ("1".equals(getNavigation()) && getPageContents().getIsNoParts())
+             && (getFeedbackComponent().getShowImmediate() || feedbackOnDate);
```

Comment:
Classic 3 line change that takes hours of investigation, cannot remember the number of times I reviewed this line!

Good job @hornersa, I'm not against the fix, maybe it's a good time to split the condition into small pieces and add some comments.

For example:

// Comment
boolean isValidAction = StringUtils.isEqualsIgnoreCase(getActionString(), "reviewAssessment", "takeAssessment", "takeAssessmentViaUrl", "previewAssessment");
// Comment
boolean isFeedbackAvailableByDate = getFeedbackComponent().getShowImmediate() || feedbackOnDate;
// Comment
boolean isNoParts = ! ("1".equals(getNavigation()) && getPageContents().getIsNoParts());

return isValidAction && isNoParts  && isFeedbackAvailableByDate;

Reply:
@mpellicer - I'm wondering if we can omit the "isNoParts" boolean entirely. I'm struggling to identify a case where getPageContents().getIsNoParts() returns true. I'm not aware of how to create an assessment in T&Q that does not have one or more parts. Are you?